pkg/reconciler/apis/apiexport/apiexport_controller.go

```diff
@@ -322,10 +323,6 @@ func (c *controller) patchIfNeeded(ctx context.Context, old, obj *apisv1alpha1.A
 	specOrObjectMetaChanged := !equality.Semantic.DeepEqual(old.Spec, obj.Spec) || !equality.Semantic.DeepEqual(old.ObjectMeta, obj.ObjectMeta)
 	statusChanged := !equality.Semantic.DeepEqual(old.Status, obj.Status)
 
-	if specOrObjectMetaChanged && statusChanged {
```

Comment:
Let's put as a comment on the inner `apiExportForPatch` declaration that the logic will guarantee that only the objectmeta/spec change will be patched first and on the next iteration the spec change.